tools/nasparse.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/python3
+import pycrate_mobile
+from pycrate_mobile import NASLTE
+import pycrate_core
+import binascii
+import sys
+import pprint
+from enum import Enum
+
+import pycrate_mobile.TS24301_EMM
+
+def parse_nas_message(buffer, uplink=None):
+    bin = binascii.unhexlify(buffer)
+    if uplink:
+        parsed = NASLTE.parse_NASLTE_MO(bin)
+    elif uplink == None: #We don't know if its an up or downlink
+        parsed = NASLTE.parse_NASLTE_MO(bin)
+        if parsed[0] == None:
+            parsed = NASLTE.parse_NASLTE_MT(bin)
+    else:
+        parsed = NASLTE.parse_NASLTE_MT(bin)
+    return parsed[0]
+
+def heur_ue_imsi_sent(msg):
```

Comment:
instead of returning `(bool, string | None)` why not just return `string | None`?

Reply:
I just feel like returning a bool is cleaner especially with the above rewrite